pallets/time-release/src/migration/v2.rs

```diff
@@ -0,0 +1,278 @@
+use crate::{types, BalanceOf, BlockNumberFor, Config, FreezeReason, Pallet, ReleaseSchedules};
+use frame_support::{
+	pallet_prelude::{GetStorageVersion, IsType, Weight},
+	traits::{
+		fungible::MutateFreeze, Get, LockIdentifier, LockableCurrency, OnRuntimeUpgrade,
+		StorageVersion,
+	},
+};
+use parity_scale_codec::Encode;
+use sp_core::hexdisplay::HexDisplay;
+use sp_runtime::{traits::Zero, Saturating};
+
+const LOG_TARGET: &str = "runtime::time-release";
+
+#[cfg(feature = "try-runtime")]
+use sp_std::vec::Vec;
+
+const RELEASE_LOCK_ID: LockIdentifier = *b"timeRels";
+
+/// The OnRuntimeUpgrade implementation for this storage migration
+pub struct MigrationToV2<T, OldCurrency>(sp_std::marker::PhantomData<(T, OldCurrency)>);
+impl<T, OldCurrency> MigrationToV2<T, OldCurrency>
+where
+	T: Config,
+	OldCurrency: 'static + LockableCurrency<T::AccountId, Moment = BlockNumberFor<T>>,
+	OldCurrency::Balance: IsType<BalanceOf<T>>,
+{
+	/// Translate time release locked deposit to frozen deposit
+	pub fn translate_lock_to_freeze(
+		account_id: &T::AccountId,
+		amount: OldCurrency::Balance,
+	) -> Weight {
+		// 1 read get Freezes: set_freeze: get locks
+		// 1 read get Locks: update_freezes: Locks::get().iter()
+		// 1 write set Account: update_freezes->mutate_account->try_mutate_account->ensure_upgraded: if account is *not* already upgraded.
+		// 1 write set Account: update_freezes->mutate_account->try_mutate_account: set account data
+		// 1 write set Freezes: update_freezes: Freezes::insert
+		T::Currency::set_freeze(
+			&FreezeReason::TimeReleaseVesting.into(),
+			&account_id,
+			amount.into(),
+		)
+		.unwrap_or_else(|err| {
+			log::error!(
+				target: LOG_TARGET,
+				"Failed to freeze {:?} from account 0x{:?}, reason: {:?}",
+				amount,
+				HexDisplay::from(&account_id.encode()),
+				err
+			);
+		});
+
+		// 1 read get Locks: remove_lock: get locks
+		// 1 read get Freezes
+		// 1 read get Account
+		// 1 write set Account: update_locks->try_mutate_account->ensure_upgraded: if account is *not* already upgraded.
+		// 1 write set Account: update_locks->try_mutate_account: set account data
+		// [Cached] 1 read get Locks: update_locks: set existed with `contains_key`
+		// 1 write set Locks: update_locks->Locks::remove: remove existed
+		OldCurrency::remove_lock(RELEASE_LOCK_ID, &account_id);
+
+		T::DbWeight::get().reads_writes(5, 6)
+	}
+}
+
+impl<T: Config, OldCurrency> OnRuntimeUpgrade for MigrationToV2<T, OldCurrency>
+where
+	T: Config,
+	OldCurrency: 'static + LockableCurrency<T::AccountId, Moment = BlockNumberFor<T>>,
+	OldCurrency::Balance: IsType<BalanceOf<T>>,
+{
+	fn on_runtime_upgrade() -> Weight {
+		log::info!(target: LOG_TARGET, "Running storage migration...");
+
+		let on_chain_version = Pallet::<T>::on_chain_storage_version();
+
+		// storage was already migrated.
+		if on_chain_version.ge(&2) {
+			log::info!(
```

Comment:
nit: we should do a `warn` here than info

Reply:
I'll change it, if there are other updates.